app/controllers/meals_controller.rb

```diff
@@ -0,0 +1,23 @@
+class MealsController < ApplicationController
+  def create
+    form = MealForm.new(menu.meals.new)
+
```

Comment:
Why don't you extract meals creation & validation to a service / operation object?

`if / else` for flash messages everywhere looks ugly. 
Take a look at this file - https://github.com/pivorakmeetup/pivorak-web-app/blob/development/app/controllers/admin/events_controller.rb

Can you implement it in the same fashion?
https://github.com/pivorakmeetup/pivorak-web-app/blob/9b331e4d43a341ee887ec4f72c0aca26d127b0d2/app/controllers/admin/base_controller.rb